clang/lib/CIR/Dialect/IR/CIRTypes.cpp

```diff
@@ -289,6 +313,18 @@ void StructType::computeSizeAndAlignment(
   unsigned numElements = getNumElements();
   auto members = getMembers();
 
+  // For unions, the size and aligment is that of the largest element.
+  if (isUnion()) {
```

Comment:
Shouldn't this be computed lazily like we do for other alignment and size things? Why do we need this to be computed during parsing time?

Reply:
Yea, I got confused by the diff and thought this was done inside `::parse`, so you are already doing what I suggest :)